---
title: "Untitled"
author: "LW"
date: "7/3/2022"
output: html_document

---



<pre class="r"><code>library(Seurat)</code></pre>
<pre><code>## Registered S3 method overwritten by &#39;spatstat.core&#39;:
##   method          from
##   formula.glmmPQL MASS</code></pre>
<pre><code>## Attaching SeuratObject</code></pre>
<pre><code>## Attaching sp</code></pre>
<pre class="r"><code>library(tidyverse)</code></pre>
<pre><code>## ── Attaching packages ─────────────────────────────────────── tidyverse 1.3.1 ──</code></pre>
<pre><code>## ✔ ggplot2 3.3.6     ✔ purrr   0.3.4
## ✔ tibble  3.1.7     ✔ dplyr   1.0.9
## ✔ tidyr   1.2.0     ✔ stringr 1.4.0
## ✔ readr   2.1.2     ✔ forcats 0.5.1</code></pre>
<pre><code>## ── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
## ✖ dplyr::filter() masks stats::filter()
## ✖ dplyr::lag()    masks stats::lag()</code></pre>
<pre class="r"><code>library(future)
plan()</code></pre>
<pre><code>## sequential:
## - args: function (..., envir = parent.frame())
## - tweaked: FALSE
## - call: NULL</code></pre>
<pre class="r"><code>plan(&quot;multisession&quot;, workers = 16)
plan()</code></pre>
<pre><code>## multisession:
## - args: function (..., workers = 16, envir = parent.frame())
## - tweaked: TRUE
## - call: plan(&quot;multisession&quot;, workers = 16)</code></pre>
<pre class="r"><code>options(future.globals.maxSize =1024*1024^2)

sham1 &lt;- ReadMtx(
  mtx=&quot;./sham/GSM5319987_sham1_matrix.mtx.gz&quot;,
  cells = &quot;./sham/GSM5319987_sham1_barcodes.tsv.gz&quot;,
  features = &quot;./sham/GSM5319987_sham1_genes.tsv.gz&quot;,
  cell.column = 1,
  feature.column = 2,
  cell.sep = &quot;\t&quot;,
  feature.sep = &quot;\t&quot;,
  skip.cell = 0,
  skip.feature = 0,
  mtx.transpose = FALSE,
  unique.features = TRUE,
  strip.suffix = FALSE
)
#Create seurat object
sham1 &lt;- CreateSeuratObject(counts = sham1)
sham1$condition &lt;- &quot;sham&quot;
#Calculate mitochondrial gene ratio
sham1$percent_mt &lt;- PercentageFeatureSet(sham1, pattern = &quot;^mt-&quot;)
#Calculate ribosomal gene ratio
# https://kb.10xgenomics.com/hc/en-us/articles/218169723-What-fraction-of-reads-map-to-ribosomal-proteins-
sham1$percent_rb &lt;- PercentageFeatureSet(sham1, pattern = &quot;^Rp[sl]&quot;)
rownames(sham1[grep(pattern = &quot;Ercc&quot;, x = rownames(sham1)),])</code></pre>
<pre><code>## [1] &quot;Ercc5&quot;   &quot;Ercc6l&quot;  &quot;Ercc1&quot;   &quot;Ercc2&quot;   &quot;Ercc6&quot;   &quot;Ercc6l2&quot; &quot;Ercc8&quot;  
## [8] &quot;Ercc4&quot;   &quot;Ercc3&quot;</code></pre>
<pre class="r"><code>VlnPlot(sham1, features = c(&quot;nFeature_RNA&quot;, &quot;nCount_RNA&quot;, &quot;percent_mt&quot;, &quot;percent_rb&quot;), ncol = 4)</code></pre>
<p><img src="/post/test_GSE/CopyOfUntitled_files/figure-html/unnamed-chunk-1-1.png" width="672" /></p>
