---
title: "Untitled"
author: "LW"
date: "7/3/2022"
output: html_document

---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r}
library(Seurat)
library(tidyverse)
library(future)
plan()
plan("multisession", workers = 16)
plan()
options(future.globals.maxSize =1024*1024^2)

sham1 <- ReadMtx(
  mtx="./sham/GSM5319987_sham1_matrix.mtx.gz",
  cells = "./sham/GSM5319987_sham1_barcodes.tsv.gz",
  features = "./sham/GSM5319987_sham1_genes.tsv.gz",
  cell.column = 1,
  feature.column = 2,
  cell.sep = "\t",
  feature.sep = "\t",
  skip.cell = 0,
  skip.feature = 0,
  mtx.transpose = FALSE,
  unique.features = TRUE,
  strip.suffix = FALSE
)
#Create seurat object
sham1 <- CreateSeuratObject(counts = sham1)
sham1$condition <- "sham"
#Calculate mitochondrial gene ratio
sham1$percent_mt <- PercentageFeatureSet(sham1, pattern = "^mt-")
#Calculate ribosomal gene ratio
# https://kb.10xgenomics.com/hc/en-us/articles/218169723-What-fraction-of-reads-map-to-ribosomal-proteins-
sham1$percent_rb <- PercentageFeatureSet(sham1, pattern = "^Rp[sl]")
rownames(sham1[grep(pattern = "Ercc", x = rownames(sham1)),])
VlnPlot(sham1, features = c("nFeature_RNA", "nCount_RNA", "percent_mt", "percent_rb"), ncol = 4)

```

